src/cluster.h

```diff
@@ -337,7 +338,8 @@ typedef struct {
     char slaveof[CLUSTER_NAMELEN];
     char myip[NET_IP_STR_LEN];    /* Sender IP, if not all zeroed. */
     uint16_t extensions; /* Number of extensions sent along with this packet. */
-    char notused1[30];   /* 30 bytes reserved for future usage. */
+    uint16_t extensions_supported; /* Does the node support parsing ping extensions. */
```

Comment:
This is fine, except it maybe should be like mflags2 or something. I would use 8 bits and not 16.

Reply:
Didn't intend for this variable to be used for any other purpose. Do we want it to renamed to `mflags2` ?

Modified it to 8 bits.

Reply:
Oops, there is some issue with macos on modifying it to 8 bits. 

```
cluster_legacy.c:3614:33: error: implicit conversion from '__uint16_t' (aka 'unsigned short') to 'uint8_t' (aka 'unsigned char') changes value from 256 to 0 [-Werror,-Wconstant-conversion]
    hdr->extensions_supported = htons(1);
                              ~ ^~~~~~~~
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/sys/_endian.h:133:25: note: expanded from macro 'htons'
#define htons(x)        __DARWIN_OSSwapInt16(x)
                        ^~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode_14.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/libkern/_OSByteOrder.h:85:6: note: expanded from macro '__DARWIN_OSSwapInt16'
    ((__uint16_t)(__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt16(x) : _OSSwapInt16(x)))
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.

```

Reply:
The API(s) are the following:
```
extern uint16_t ntohs (uint16_t __netshort)
     __THROW __attribute__ ((__const__));
extern uint16_t htons (uint16_t __hostshort)
     __THROW __attribute__ ((__const__));
```

I would rather keep the variable as uint16_t` to avoid any issue during conversion.